awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -229,14 +230,24 @@ def _expand_path(self, path):
 
 
 class EKSClient(object):
-    def __init__(self, session, cluster_name, role_arn, parsed_globals=None):
+    # A set that contains optional flag keys here. 
+    # If you need to specify a default value other than `None` for the parsed_args,
+    # you need to provide a value for the 'default' key in ARG_TABLE. 
+    _OPTIONAL_ATTRS = {
+        'role_arn',
+    }
+
+    def __init__(self, session, *, cluster_name, parsed_globals=None, **parsed_args):
         self._session = session
-        self._cluster_name = cluster_name
-        self._role_arn = role_arn
         self._cluster_description = None
+        self._cluster_name = cluster_name
         self._globals = parsed_globals
+        
+        for name in EKSClient._OPTIONAL_ATTRS:
+            setattr(self, f'_{name}', parsed_args[name])
 
-    def _get_cluster_description(self):
+    @property
+    def cluster_description(self):
```

Comment:
I would also add a [`@functools.lru_cache`](https://docs.python.org/3.7/library/functools.html#functools.lru_cache) decoration here, otherwise it'll be an expensive amount of EKS client calls for every operation.